src/svg/helper/PatternManager.ts

```diff
@@ -122,7 +122,7 @@ export default class PatternManager extends Definable {
      * @param pattern zr pattern instance
      * @param patternDom DOM to update
      */
-    updateDom(pattern: PatternObject, patternDom: SVGElement) {
+    updateDom(pattern: SVGPatternObject, patternDom: SVGElement) {
```

Comment:
Sorry for the confusing name `CanvasPatterObject`. The difference between these two pattern object is whether to use an image in the pattern or use an SVG element. Both of them are supported in SVG renderer. https://github.com/ecomfe/zrender/blob/d30bcf75748ffb40cec43794fddcf89c88660258/src/svg/helper/PatternManager.ts#L139

So perhaps using `ImagePatternObject` and `SVGPatternObject` is clearer.

Reply:
Not sure if 3c2fb9a understands what you mean.

Reply:
And SVGPatternObject should not have `image` property.